### ISSUE TITLE
Fill in the blank option

### ISSUE THREAD

**Issue by ascholz81** [ResearchKit/ResearchKit#548]
I have survey with multiple choice questions.  Several of those options include an "Other: __________ " (Fill in the blank) option.  Currently I'm using a navigation rule to detect that you selected "Other" to present a follow up question with an ORKTextAnswerFormat to gather that data.

Is there another way to do this?  Perhaps a new feature if there isn't a way?

**Comment by YuanZhu-apple** [comment]
You can try `ORKFormStep` with a choice selection and a text view, even though there is no validation rule to enforce text to be filled when `Other` is picked. 

Please let us know which kind of enforcement rule do you need.

**Comment by md0u80c9** [comment]
Yes, you can do this with an ORKFormStep and the navigation rules as YuanZhu describes.

Step 1:
What is your favourite animal?
Cats
Dogs
Rabbits
Other

Step 2
What animal do you prefer?
Animal: [Text field]

Step 3 (Completion step):
Thanks. 

Then I would use ORKResultPredicates to set the different navigation paths up. Set up a predicate so that if Step 1 chooses Dogs or Cats or Rabbits then go to step 3.

You can conversely (depending upon the complexity of the answer), swap steps 2 and 3 so that the 'end' step is Step 2. Then you need a predicate to 'Other' to go to step 3, and an ORKDirectStepNavigationRule to "" on Step 2.

**Comment by ascholz81** [comment]
Andrew:  Do you mean this to be separate steps? I first read your response as this could all be contained in one form step, however upon re-reading and trial/error I see this was probably not your intent.  Just wanted to confirm.
  If it is separate steps, that is what I am currently doing.

**Comment by md0u80c9** [comment]
Yes, I meant do it in separate steps but use a predicate so that if you pick anything other than 'Other' that you don't have to enter Step 2. That way the majority of your clients see the bare minimum options.

**Comment by ascholz81** [comment]
Yuan:  An enforcement rule could be applied via a regEx or by a character count.  However, I would say generically not blank is the best enforcement rule since "Other" covers "everything else."  In my situation I just need to know it isn't blank.

**Comment by YuanZhu-apple** [comment]
@ascholz81 Do you mean if `Other` was picked, then the user cannot leave the additional text field blank .
So the enforcement rule is conditional, decide by which choice is picked.

**Comment by ascholz81** [comment]
Correct.

**Comment by YuanZhu-apple** [comment]
Mockup screen:

<img width="378" alt="screen shot 2015-11-03 at 4 18 19 pm" src="https://cloud.githubusercontent.com/assets/11466704/10925870/c1bacde2-8246-11e5-8f2d-d4b5d12b7f39.png">

**Comment by md0u80c9** [comment]
Yes that would be nice Yuan.

**Comment by ascholz81** [comment]
Very nice.

**Comment by Pariecemckinney-apple** [comment]
This functionality is available today via the `ORKTextChoiceOther` class that can now be passed in with other text choices to a `ORKTextChoiceAnswerFormat`.  

 Closing issue. Please feel free to open another issue if further discussion is needed.
